app/models/manageiq/providers/kubevirt/infra_manager/metrics_capture/prometheus_capture_context.rb

```diff
@@ -70,10 +70,10 @@ def labels_to_s(labels)
   end
 
   def validate_target
-    raise TargetValidationError, "ems not defined" unless @ext_management_system
-    raise TargetValidationWarning, "no associated hardware" unless @vm_hardware
+    raise ManageIQ::Providers::Kubevirt::InfraManager::MetricsCapture::TargetValidationError, "ems not defined" unless @ext_management_system
+    raise ManageIQ::Providers::Kubevirt::InfraManager::MetricsCapture::TargetValidationWarning, "no associated hardware" unless @vm_hardware
 
-    raise TargetValidationError, "cores not defined" unless @vm_cores.to_i > 0
-    raise TargetValidationError, "memory not defined" unless @vm_memory.to_i > 0
```

Comment:
This was needed since the capture context is subclassed from k8s which has its own error handling classes
```
class ManageIQ::Providers::Kubevirt::InfraManager::MetricsCapture::PrometheusCaptureContext < ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture::PrometheusCaptureContext
```

This should not be needed if https://github.com/ManageIQ/manageiq-providers-kubernetes/pull/543 is resolved but that may necessitate other changes to the specs